Dart/src/com/jetbrains/lang/dart/ide/actions/DartPubActionBase.java

```diff
@@ -178,6 +178,8 @@ private static void doPerformPubAction(@NotNull final Module module,
 
     try {
       if (ourInProgress.compareAndSet(false, true)) {
+        command.withEnvironment("PUB_ENVIRONMENT", "IntelliJ-IDEA");
```

Comment:
I think we want to use what we use for the Analysis Server:

myServerSocket.setClientId(ApplicationNamesInfo.getInstance().getFullProductName().replace(' ', '_'));

Perhaps the DartAnalysisServerService should have a static method getClientId which returns ApplicationNamesInfo.getInstance().getFullProductName().replace(' ', '_') for this location, the location below and for the usage in the analysis server?

Perhaps we should call a method